cmd/node/main.go

```diff
@@ -1115,9 +1138,48 @@ func startNode(ctx *cli.Context, log logger.Logger, version string) error {
 	err = networkComponents.NetMessenger.Close()
 	log.LogIfError(err)
 
+	handleAppClose(log, sig)
+
 	return nil
 }
 
+func handleAppClose(log logger.Logger, endProcessArgument endProcess.EndProcessArgument) {
+	log.Debug(fmt.Sprintf("Restarting because %s. Details: %s",
+		endProcessArgument.Reason,
+		endProcessArgument.Description))
+	switch endProcessArgument.Reason {
+	case core.ShuffledOut:
+		{
+			newStartInEpoch(log)
+		}
+	}
+}
+
+func newStartInEpoch(log logger.Logger) {
+	wd, err := os.Getwd()
+	if err != nil {
+		log.LogIfError(err)
+	}
+	nodeApp := os.Args[0]
+	args := os.Args
+	args = append(args, "-start-in-epoch")
+
+	log.Debug("Executable", "working dir", wd)
```

Comment:
Why not directly one line: log.Debug("Executable", "working dir", wd, "node app", nodeApp, "args", args) and remove lines 1168 and 1169

Reply:
fixed